Counterexamples/EulerSumOfPowers.lean

```diff
@@ -0,0 +1,345 @@
+/-
+Copyright (c) 2025 Snir Broshi. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Snir Broshi
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset.Defs
+import Mathlib.Data.Finset.Card
+import Mathlib.Data.PNat.Notation
+import Mathlib.NumberTheory.FLT.Basic
+
+/-!
+# Euler's sum of powers conjecture
+
+Euler's sum of powers conjecture says that at least n nth powers of positive integers
+are required to sum to an nth power of a positive integer.
+
+This was an attempt to generalize Fermat's Last Theorem,
+which is the special case of summing 2 nth powers.
+
+https://en.wikipedia.org/wiki/Euler%27s_sum_of_powers_conjecture
+http://euler.free.fr/
+-/
+
+
+namespace Counterexample
+
+abbrev sumOfPowersConjectureAux {α β γ : Type*}
+    [Zero α] [Zero β] [Zero γ] [Add γ] [HPow α ℕ γ] [HPow β ℕ γ] :=
```

Comment:
I find it very unlikely that Euler stated this question about arbitrary power operators of type `α -> ℕ -> γ`; what would you lose by merging these three types into one type and assumint `Semiring α`?

Reply:
Also, I'd suggest to call the semiring `R` and make it explicit; then you can write `SumOfPowersConjecture ℕ` and don't need the `abbrev`s below.

Reply:
Further, note that your version is trivially false for any nontrivial semiring:
```lean
```def sumOfPowersConjecture' (R : Type*) [Semiring R] : Prop :=
  ∀ (n : ℕ) (a : List R) (b : R),
  0 ∉ a → b ≠ 0 → (a.map (· ^ n)).sum = b ^ n → n ≤ a.length

example (R : Type*) [Semiring R] [Nontrivial R] : ¬ sumOfPowersConjecture' R := by
  intro h
  have : 2 ≤ 1 := h 2 [1] 1 (by simp) (by norm_num) (by simp)
  omega
```
You are missing a conidition like `2 ≤ a.length` (the coditions `n ≠ 0` and `a ≠ ∅` are redundant in your formulation: when `n = 0`, the conclusion is trivially true, and when `a` is empty, the sum will be zero, which is excluded (in reasonable semirings)  by `b ≠ 0`. In any case, in the corrected form, `2 ≤ a.length` implies it-)